src/main/java/org/springframework/data/elasticsearch/core/JestElasticsearchTemplate.java

```diff
@@ -0,0 +1,1283 @@
+			for (SortBuilder sort : searchQuery.getElasticsearchSorts()) {
+				searchSourceBuilder.sort(sort);
+			}
+		}
+
+		if (searchQuery.getHighlightFields() != null) {
+			HighlightBuilder highlighter = searchSourceBuilder.highlighter();
+			for (HighlightBuilder.Field highlightField : searchQuery.getHighlightFields()) {
+				highlighter.field(highlightField);
+			}
+		}
+
+		if (!isEmpty(searchQuery.getAggregations())) {
+			for (AbstractAggregationBuilder aggregationBuilder : searchQuery.getAggregations()) {
+				searchSourceBuilder.aggregation(aggregationBuilder);
+			}
+		}
+
+		if (!isEmpty(searchQuery.getIndicesBoost())) {
+			for (IndexBoost indexBoost : searchQuery.getIndicesBoost()) {
+				searchSourceBuilder.indexBoost(indexBoost.getIndexName(), indexBoost.getBoost());
+			}
+		}
+
+		if (!searchQuery.getScriptFields().isEmpty()) {
+			searchSourceBuilder.field("_source");
+			for (ScriptField scriptedField : searchQuery.getScriptFields()) {
+				searchSourceBuilder.scriptField(scriptedField.fieldName(), scriptedField.script());
+			}
+		}
+
+		return executeSearch(searchQuery, searchSourceBuilder.query(searchQuery.getQuery()));
+	}
+
+	private SearchResult executeSearch(Query query, SearchSourceBuilder request) {
+
+		Search.Builder search = new Search.Builder(request.toString());
+		if (query != null) {
+			search.
+					addType(query.getTypes()).
+					addIndex(query.getIndices()).
+					setSearchType(SearchType.valueOf(query.getSearchType().name()));
+		}
+
+		return new ExtendedSearchResult(execute(search.build()));
+	}
+
+	private Index prepareIndex(IndexQuery query) {
+		try {
+			String indexName = isBlank(query.getIndexName()) ? retrieveIndexNameFromPersistentEntity(query.getObject()
+					.getClass())[0] : query.getIndexName();
+			String type = isBlank(query.getType()) ? retrieveTypeFromPersistentEntity(query.getObject().getClass())[0]
+					: query.getType();
+
+			Index.Builder indexBuilder;
+
+			if (query.getObject() != null) {
+				String entityId = null;
+				if (isDocument(query.getObject().getClass())) {
+					entityId = getPersistentEntityId(query.getObject());
+				}
+
+				indexBuilder = new Index.Builder(resultsMapper.getEntityMapper().mapToString(query.getObject()));
+
+				// If we have a query id and a document id, do not ask ES to generate one.
+				if (query.getId() != null && entityId != null) {
+					indexBuilder.index(indexName).type(type).id(query.getId());
+				} else {
+					indexBuilder.index(indexName).type(type);
+				}
+			} else if (query.getSource() != null) {
+				indexBuilder = new Index.Builder(query.getSource()).index(indexName).type(type).id(query.getId());
+			} else {
+				throw new ElasticsearchException("object or source is null, failed to index the document [id: " + query.getId() + "]");
+			}
+			if (query.getVersion() != null) {
+				indexBuilder.setParameter(Parameters.VERSION, query.getVersion());
+				indexBuilder.setParameter(Parameters.VERSION_TYPE, EXTERNAL);
```

Comment:
Elasticsearch expects `version_type` parameter value to be lowercase (case-sensitive), i.e. it should be "external", not "EXTERNAL" . Suggestion: use `EXTERNAL.name().toLowerCase()`.